src/sagemaker/jumpstart/utils.py

```diff
@@ -810,3 +829,74 @@ def get_jumpstart_model_id_version_from_resource_arn(
             model_version = model_version_from_tag
 
     return model_id, model_version
+
+
+def extract_info_from_hub_content_arn(
+    arn: str,
+) -> HubArnExtractedInfo:
+    """Extracts descriptive information from a Hub or HubContent Arn"""
+
+    match = re.match(constants.HUB_CONTENT_ARN_REGEX, arn)
+    if match:
+        partition = match.group(1)
+        hub_region = match.group(2)
+        account_id = match.group(3)
+        hub_name = match.group(4)
+        hub_content_type = match.group(5)
+        hub_content_name = match.group(6)
+        hub_content_version = match.group(7)
+
+        return HubArnExtractedInfo(
+            partition=partition,
+            region=hub_region,
+            account_id=account_id,
+            hub_name=hub_name,
+            hub_content_type=hub_content_type,
+            hub_content_name=hub_content_name,
+            hub_content_version=hub_content_version,
+        )
+
+    match = re.match(constants.HUB_ARN_REGEX, arn)
```

Comment:
why is `extract_info_from_hub_content_arn` also handling hubs and not just hub content?

Reply:
I should come up with a better name, or break it up into two functions

Reply:
yeah this is an odd pattern, we can end up using regexes unnecessarily for hub arns